parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/RuleDocumentationUrlProvider.java

```diff
@@ -16,30 +16,36 @@
 
 package com.parasoft.findings.teamcity.agent;
 
-import com.parasoft.xtest.common.dtp.IDtpServiceRegistry;
-import com.parasoft.xtest.common.services.RawServiceContext;
-import com.parasoft.xtest.configuration.dtp.XRestRulesClient;
-import com.parasoft.xtest.configuration.rules.RuleDocumentationHelper;
-import com.parasoft.xtest.services.api.IParasoftServiceContext;
-import com.parasoft.xtest.services.api.ServiceUtil;
+import com.parasoft.findings.utils.doc.RuleDocumentationProvider;
+import com.parasoft.findings.utils.doc.RuleDocumentationProvider.ClientStatus;
 import jetbrains.buildServer.agent.AgentRunningBuild;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Properties;
 
+import static com.parasoft.findings.teamcity.agent.LocalSettingsHelper.DTP_URL;
+
 public class RuleDocumentationUrlProvider {
     private static AgentRunningBuild _build;
     private Map<String, String> _ruleDocsUrls;
-    private final TeamCityRuleDocumentationProvider _docProvider;
+    private final RuleDocumentationProvider _docProvider;
 
-    public RuleDocumentationUrlProvider(AgentRunningBuild build, Properties settings) {
+    public RuleDocumentationUrlProvider(AgentRunningBuild build, Properties settings ) {
         _build = build;
         _ruleDocsUrls = new HashMap<>();
-        _docProvider = new TeamCityRuleDocumentationProvider(settings);
+        _docProvider = new RuleDocumentationProvider(settings);
+        if (_docProvider.getDtpDocServiceStatus() == ClientStatus.NOT_AVAILABLE) {
```

Comment:
Is it better to add a string field for description of error? In that way we don't need to write the message in different projects.
```
public enum ClientStatus {
    AVAILABLE("DTP server is available"),
    NOT_SUPPORTED_VERSION("Unable to retrieve the documentation for the rule from DTP. Please ensure that a DTP version older than 2023.1 is used"),
    NOT_AVAILABLE("DTP server is not available"),
    DTP_URL_NOT_SPECIFIED("DTP URL is not specified");

    private final String description;

    ClientStatus(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return description;
    }
}

```